gatherers/service-worker.js

```diff
@@ -20,11 +20,15 @@ const Gather = require('./gather');
 
 class ServiceWorker extends Gather {
   setup(options) {
+    this.resolved = false;
+
     this.artifactsResolved = new Promise((res, _) => {
       options.driver.on(
           'ServiceWorker.workerVersionUpdated', data => {
-            if (ServiceWorker.getActivatedServiceWorker(data.versions) !== undefined) {
+            if (ServiceWorker.getActivatedServiceWorker(data.versions) !== undefined &&
+                !this.resolved) {
               this.artifact = {serviceWorkers: data};
+              this.resolved = true;
```

Comment:
Since artifacts is a object, when we override a prop on it, it propagates to all references of that object. So since we are not calling off on the event, we were resolving the promise and then changing the underlying prop on artifacts after. Which caused us problems.